ROADMAP.md

```diff
@@ -0,0 +1,52 @@
+# SkyPilot Roadmap
+
+## v0.3
```

Comment:
I would like to call this `v1.0.0-dev0` or `v1.0.0` to align with the current master version. Reason: I believe the `v0.3` will not include all the features listed here and we should not make a promise for these features to be included in `v0.3`. This doc is a longer-term plan than `v0.3`.